ehitchhiking-ui/src/app/pages/main-screen/main-screen-component/main-screen.component.ts

```diff
@@ -8,5 +10,57 @@ import {Component, OnInit} from '@angular/core';
 export class MainScreenComponent implements OnInit {
 	constructor() {}
 
-	ngOnInit() {}
+	tripFormData: any; // TODO
+	isHiddenTripRegistration: boolean;
+	userState: string;
+	isSavedRoute: boolean;
+	isShownRoutesList: boolean;
+	isShownViewRoutesButton: boolean;
+	isShownSaveRouteButton: boolean;
+
+	routes: Partial<Route>[] = [];
+
+	ngOnInit() {
+		this.isHiddenTripRegistration = true;
+		this.userState = UserState.passenger;
+		this.routes = YandexMapService.getSomeRoutes();
+	}
+
+	openTripRegistrationForm(): void {
+		this.isHiddenTripRegistration = !this.isHiddenTripRegistration;
+	}
+
+	getData(data) {
+		this.tripFormData = data;
+    this.isHiddenTripRegistration = true;
+	}
+
+	saveRoute() {
+		this.isSavedRoute = !this.isSavedRoute;
+		this.isShownSaveRouteButton = false;
+	}
+
+	viewRoutes() {
+		this.isShownRoutesList = !this.isShownRoutesList;
+	}
+
+	getIsShown(data) {
```

Comment:
method name starts with get but method doesn't return anything. pls rename to setIsShownViewRoutesButtonFlag

Reply:
fixed